http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/Configuration.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2010-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.nio.netty.internal;
+
+import static software.amazon.awssdk.http.SdkHttpConfigurationOption.CONNECTION_TIMEOUT;
+import static software.amazon.awssdk.http.SdkHttpConfigurationOption.MAX_CONNECTIONS;
+import static software.amazon.awssdk.http.SdkHttpConfigurationOption.USE_STRICT_HOSTNAME_VERIFICATION;
+import static software.amazon.awssdk.utils.NumericUtils.saturatedCast;
+
+import java.time.Duration;
+import java.util.Optional;
+import software.amazon.awssdk.annotations.Immutable;
+import software.amazon.awssdk.annotations.ReviewBeforeRelease;
+import software.amazon.awssdk.http.nio.netty.NettySdkHttpClientFactory;
+import software.amazon.awssdk.utils.AttributeMap;
+
+/**
+ * Helper class to unwrap and convert service defaults.
+ */
+@Immutable
+public final class Configuration {
```

Comment:
More specific name maybe? We have lots of configuration classes

Reply:
Yeah - that's fair, I thought I  might be able to get away with it because it's internal in the netty package, but you're right we do have a few configuration classes. I actually want to re-visit the whole way we configure HTTP clients because in doing this, I found myself having to thread new values through many classes - which was a bit of a smell.